src/components/App.tsx

```diff
@@ -900,6 +904,13 @@ class App extends React.Component<ExcalidrawProps, AppState> {
         gridSize: this.props.gridModeEnabled ? GRID_SIZE : null,
       });
     }
+
+    if (prevProps.name !== this.props.name && this.props.name) {
```

Comment:
```suggestion
    if (this.props.name && prevProps.name !== this.props.name) {
```
gotcha, lets move that check in the beginning as the comparison need not be done if not present.